### PR TITLE
Adding ServiceNow ITSM and CSM Templates

### PR DESCRIPTION
**Pull request by shash-sgnl**
https://app.shortcut.com/sgnl/story/15629/feature-request-disney-support-for-servicenow-s-incident-and-problem-management-components

Added Incidents, ChangeRequest and Problems in the ITSM template.

### CODE REVIEW

**Comment by shash-sgnl** [comment]
@joewelsh please review keeping in mind Disney's use-cases. Please let me know if we need to add support for anything else.

File: servicenow-itsm.sgnl.yaml [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+entities:
+  User:
+    displayName: User
+    externalId: sys_user
+    description: User Entity in ServiceNow
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 999
```
Comment (marc-sgnl) [inline servicenow-itsm.sgnl.yaml:28]: @joewelsh would love your feedback on Disney's config here. @shash-sgnl for context, we found that the SNOW API was timing out at this page size. I'd recommend something between 250 and 400 (for all entities) based on my experience testing at Disney. But Joe can provide first-hand feedback here too.

File: servicenow-csm.sgnl.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # ServiceNow Configuration YAML for DS2.0
```
Comment (marc-sgnl) [inline servicenow-csm.sgnl.yaml:1]:
```suggestion
# ServiceNow CSM Configuration YAML for DS2.0
```

File: servicenow-itsm.sgnl.yaml [hunk 3]
```diff
@@ -0,0 +1,689 @@
+# ServiceNow Configuration YAML for DS2.0
```
Comment (marc-sgnl) [inline servicenow-itsm.sgnl.yaml:1]:
```suggestion
# ServiceNow ITSM Configuration YAML for DS2.0
```

File: servicenow-itsm.sgnl.yaml [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+        type: String
+      - name: sysModCount
+        externalId: sys_mod_count
+        type: String
+      - name: sysTags
+        externalId: sys_tags
+        type: String
+
+  Incident:
+    displayName: Incidents
```
Comment (marc-sgnl) [inline servicenow-itsm.sgnl.yaml:361]: should be singular?

**Review by marc-sgnl — CHANGES_REQUESTED** [review]

File: servicenow-itsm.sgnl.yaml (hunk 1 above)
Comment (joewelsh) [reply servicenow-itsm.sgnl.yaml:28]: Hmm, not entirely sure why we set it that high.  Last I recall was a conversation with Romain regarding this, but could be wrong.  I'd stick with the recommended 250-400.

Comment (shash-sgnl) [reply servicenow-itsm.sgnl.yaml]: Set pagesize for all entities to 350..

**Comment by shash-sgnl** [comment]
Screenshot 2023-12-22 at 1 33 56 PM
Screenshot 2023-12-22 at 1 34 06 PM

Back in your queue @joewelsh @marc-sgnl

**Comment by joewelsh** [comment]
> [ ](/shash-sgnl)

ITSM -> Incident is missing the following attributes:
active, cmdb_ci, configurationItems, sysCreatedOn

ITSM -> ChangeRequest is missing the following attributes:
active

ITSM -> Problem
cmdb_ci, u_area, u_phase, opened_at

ITSM is missing the following entities and associated attributes:

ChangeTask (change_task)
sysId, active, assignedTo, assignmentGroup, changeTaskType, number, state

CMDBBuisnessApp (cmdb_ci_business_app)
sysId, number

You'll also need the following relationships:

Screenshot 2024-01-04 at 12 56 56 PM

Screenshot 2024-01-04 at 12 58 04 PM


@shash-sgnl the [disney-poc SGNL client](https://console.sgnl.cloud/disney-poc/systems-of-record/77e3da5f-9b58-41ff-9ff9-20d890ae8bc3/entities) uses Disney's OKTA for SSO.  I verified, you are a valid user.  Please let me know if you're unable to login to their SGNL client, I'm happy to reset password, etc.

Sorry for the delay.  Hope this is helpful?

**Comment by shash-sgnl** [comment]
> > [ ](/shash-sgnl)
> 
> ITSM -> Incident is missing the following attributes: active, cmdb_ci, configurationItems, sysCreatedOn
> 
> ITSM -> ChangeRequest is missing the following attributes: active
> 
> ITSM -> Problem cmdb_ci, u_area, u_phase, opened_at
> 
> ITSM is missing the following entities and associated attributes:
> 
> ChangeTask (change_task) sysId, active, assignedTo, assignmentGroup, changeTaskType, number, state
> 
> CMDBBuisnessApp (cmdb_ci_business_app) sysId, number
> 
> You'll also need the following relationships:
> 
> Screenshot 2024-01-04 at 12 56 56 PM Screenshot 2024-01-04 at 12 58 04 PM
> @shash-sgnl the [disney-poc SGNL client](https://console.sgnl.cloud/disney-poc/systems-of-record/77e3da5f-9b58-41ff-9ff9-20d890ae8bc3/entities) uses Disney's OKTA for SSO. I verified, you are a valid user. Please let me know if you're unable to login to their SGNL client, I'm happy to reset password, etc.
> 
> Sorry for the delay. Hope this is helpful?

@joewelsh Please review; I've added Change task and the attributes you suggested. Sync working perfectly and all relationships created.

**Review by joewelsh — APPROVED** [review]
Looks great, @shash-sgnl.  Truly appreciate all the efforts.

**Pull request merged**